AK/Math.h

```diff
@@ -60,6 +57,116 @@ INTEGER_BUILTIN(clz);
+                : "st");                                                                                                          \
+        if constexpr (sizeof(I) == sizeof(i32))                                                                                   \
+            asm(                                                                                                                  \
+                "fistpl %0"                                                                                                       \
+                : "=m"(res)                                                                                                       \
+                : "t"(value)                                                                                                      \
+                : "st");                                                                                                          \
+        if constexpr (sizeof(I) == sizeof(i16))                                                                                   \
+            asm(                                                                                                                  \
+                "fistps %0"                                                                                                       \
+                : "=m"(res)                                                                                                       \
+                : "t"(value)                                                                                                      \
+                : "st");                                                                                                          \
+        /* restore the controll word*/                                                                                            \
+        asm("fldcw %0" ::"m"(cw));                                                                                                \
+        return res;                                                                                                               \
+    }
+
+ROUNDING_FUNCTION(floor, FE_DOWNWARD);
+ROUNDING_FUNCTION_TO_INT(floor, FE_DOWNWARD);
+ROUNDING_FUNCTION(round, FE_TONEAREST);
+ROUNDING_FUNCTION_TO_INT(round, FE_TONEAREST);
+ROUNDING_FUNCTION(ceil, FE_UPWARD);
+ROUNDING_FUNCTION_TO_INT(ceil, FE_UPWARD);
+ROUNDING_FUNCTION(trunc, FE_TOWARDZERO);
+
+template<Signed I, FloatingPoint T>
+constexpr I trunc_to_int(T value) requires(sizeof(I) == sizeof(i16) || sizeof(I) == sizeof(i32) || sizeof(I) == sizeof(i64))
+{
+    CONSTEXPR_STATE(trunc, value);
+    I res;
+    if constexpr (sizeof(I) == sizeof(i64))
+        asm("fisttpq %0"
+            : "=m"(res)::"st");
+    if constexpr (sizeof(I) == sizeof(i32))
+        asm("fisttpd %0"
+            : "=m"(res)::"st");
+    if constexpr (sizeof(I) == sizeof(i16))
```

Comment:
I'm not sure about using asm from a common AK headed like this. This will break arm64 hosts (again), which will be sad for a lot of M1 MacBook users.

Reply:
If we like the (measured?) performance gains from this there should at least be some #if ARCH(I686) || ARCH(X86_64) around this inline assembly